x/gov/proposal/handler.go

```diff
@@ -0,0 +1,45 @@
+package proposal
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/gov/tags"
+)
+
+// HandleSubmit submits content and does initial deposit with provided SubmitForm
+func HandleSubmit(ctx sdk.Context, k Keeper,
```

Comment:
Should this be public?

Reply:
Yes, it is a helper function which submits and does initial deposit in one.

Reply:
Per discussion - instead of doing this we should use a singleton codec and then put this logic in the keeper